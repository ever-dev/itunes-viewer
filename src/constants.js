export const MEDIA_TYPES = [
  { value: 'movie', label: 'Movie' },
  { value: 'podcast', label: 'Pod Cast' },
  { value: 'music', label: 'Music' },
  { value: 'musicVideo', label: 'Music Video' },
  { value: 'audiobook', label: 'Audio Book' },
  { value: 'shortFilm', label: 'Short Film' },
  { value: 'tvShow', label: 'TV Show' },
  { value: 'software', label: 'Software' },
  { value: 'ebook', label: 'EBook' },
  { value: 'all', label: 'All' }
];

export const ITUNE_API_URL = 'https://itunes.apple.com';
