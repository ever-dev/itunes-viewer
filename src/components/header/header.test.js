import React from 'react';
import { mount } from 'enzyme';
import { Header } from './index';
import { MEDIA_TYPES } from '../../constants';

describe('Header', () => {
  let selectedMedia = MEDIA_TYPES[0].value;
  let mockHandler = jest.fn();
  let handler = e => {
    selectedMedia = e;
  };

  it('should render with menus', () => {
    const wrapper = mount(
      <Header
        mediaTypes={MEDIA_TYPES}
        selectedMedia={selectedMedia}
        onMediaChange={mockHandler}
      />
    );
  });

  it('should call handler when Media Change', () => {
    const wrapper = mount(
      <Header
        mediaTypes={MEDIA_TYPES}
        selectedMedia={selectedMedia}
        onMediaChange={mockHandler}
      />
    );
    wrapper.children().debug();
    wrapper
      .find(`[data-test-id='menu-movie']`)
      .first()
      .simulate('click');
    expect(mockHandler).toBeCalled();
  });

  it('should change selected menu item', () => {
    const wrapper = mount(
      <Header
        mediaTypes={MEDIA_TYPES}
        selectedMedia={selectedMedia}
        onMediaChange={handler}
      />
    );
    wrapper
      .find(`[data-test-id='menu-music']`)
      .first()
      .simulate('click');
    expect(selectedMedia).toEqual('music');
  });
});
