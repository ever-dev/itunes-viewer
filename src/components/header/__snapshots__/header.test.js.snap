// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render with menus 1`] = `ReactWrapper {}`;
